omegaconf/_utils.py

```diff
@@ -691,7 +691,12 @@ def _ensure_container(target: Any, flags: Optional[Dict[str, bool]] = None) -> A
         target = OmegaConf.create(target, flags=flags)
     elif is_structured_config(target):
         target = OmegaConf.structured(target, flags=flags)
-    assert OmegaConf.is_config(target)
+    if not OmegaConf.is_config(target):
```

Comment:
It doesn't work. We need to catch the case where `OmegaConf.is_config(target)` is true because we have passed a DictConfig/ListConfig instance to `_ensure_container`.
I've changed the `if` into an `elif` in https://github.com/omry/omegaconf/pull/568/commits/0aac680601754807a7191a298925a80d6fedf11a.